src/Uno.UWP/Storage/StorageFile.iOS.cs

```diff
@@ -89,7 +89,8 @@ public override async Task<Stream> OpenStreamAsync(CancellationToken ct, FileAcc
 			public override Task<StorageStreamTransaction> OpenTransactedWriteAsync(CancellationToken ct, StorageOpenOptions option) => throw new NotImplementedException();
 
 			protected override bool IsEqual(ImplementationBase implementation) =>
-				implementation is SecurityScopedFile file && file._nsUrl.FilePathUrl.Path == _nsUrl.FilePathUrl.Path;
+				implementation is SecurityScopedFile file && 
+				file._nsUrl.FilePathUrl?.Path == _nsUrl.FilePathUrl?.Path;
```

Comment:
@MartinZikmund If `file._nsUrl.FilePathUrl` is null, and `_nsUrl.FilePathUrl.Path` is null, we will be returning true here for `IsEqual`. But I don't know if that's what we want. (to be clear, do we want to treat a null `FilePathUrl` the same as a non-null `FilePathUrl` whose `Path` is null?)

Looks like there might not be enough test coverage here. None of the existing tests is reaching here with a null `FilePathUrl` (otherwise, it would have crashed before your change)